src/osd/pg_scrubber.cc

```diff
@@ -585,17 +585,21 @@ bool PgScrubber::write_blocked_by_scrub(const hobject_t& soid)
 	   << preemption_data.is_preemptable() << " already preempted? "
 	   << preemption_data.was_preempted() << dendl;
 
+  if (preemption_data.was_preempted()) {
+    // otherwise - write requests arriving while 'already preempted' is set
+    // but 'preemptable' is not - will not be allowed to continue, and will
+    // not be requeued on time.
+    return false;
```

Comment:
It feels dangerous to ignore m_start -> m_end blocking range, but I assume that we are guaranteeing that the current scrub chunk will be aborted.  Somewhere m_end = m_start might happen before eventually restarting that chunk or a smaller chunk at m_start.

Reply:
will verify. Thanks

Reply:
I don't see a scenario in which we have a problem with modifying 'end' in that direction (i.e. to be equal to 'start').
The current chunk will be aborted. And until that happens:
even the backend building of the local map is using its own copy of the start and end markers.